Lab 01 - Sort by Name and Age
-

Created class Person with encapsulated private fields and getters for a firstName and age.

Lab 02 - Salary Increase
-

Created a class Person with private fields and private methods. Public method for setSalary() as it will be increased 
by the Main class logic. toString() method overridden in Person as required. 

Lab 03 - Validation Data
-

Same as the last problem, with the difference that there are some conditions that are for the name, age, and salary which 
if not met then throws and exception. The exception thrown from the Person class has to be handled in the Main class by using 
try-catch block.

Lab 04 - First and Reserve Team
-

Created a class Team which have 2 fields of type List<Person> - firstTeam and reserveTeam. Validation added for the player 
to which team to be added into the settter method, if the player is under 40 y.o. will be added to the first team. If the 
player is 40 y.o. and older, will be added to the reserve team. 

Exercise 01 - Class Box
-

Created a class Box which has fields length, width, and height. It has all parameter constructor which sets the values by 
using a private setters to validate the input according to the requirements. If any of the input values is 0 or negative, 
an exception is thrown and the program execution is terminated. The exception thrown is handled with appropriate message 
in the Main method.

Exercise 02 - Animal Farm
-

Class Chicken created which has fields and methods. Some of the methods are also encapsulated according to the requirements. 
Private setters are used to validate the initial values of the input while creating an object of the class Chicken and 
appropriate exceptions are thrown. Exceptions are handled in the Main class with the required messages. 

Exercise 03 - Shopping Spree
-

Exercise 04 - Pizza Calories
-

Created class Topping, Dough, and Pizza. ToppingModifiers and DoughModifiers are stored in Enum type. Inputs have private 
setters to validate the input according to the requirements which throws suitable exceptions. The exceptions are catch 
in the Main method with appropriate messages.